src/Microsoft.AspNetCore.Hosting/Internal/HostingLoggerExtensions.cs

```diff
@@ -23,21 +23,19 @@ public static IDisposable RequestScope(this ILogger logger, HttpContext httpCont
 
         public static void RequestStarting(this ILogger logger, HttpContext httpContext)
```

Comment:
One thing that bugs me about this change is the fact that `RequestStarting` and `RequestFinished` pretty much rely on things that happen in the `HostingApplication`. Feels like those 2 methods should be moved there for proximity.